src/Akka.Logger.Serilog.Tests/SerilogFormattingSpecs.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using Akka.Configuration;
+using Akka.Event;
+using FluentAssertions;
+using Serilog;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Logger.Serilog.Tests
+{
+    public class SerilogFormattingSpecs : TestKit.Xunit2.TestKit
+    {
+        public static readonly Config Config = @"akka.loglevel = DEBUG";
+        private ILogger _serilogLogger;
+
+        private ILoggingAdapter _loggingAdapter;
+
+        public SerilogFormattingSpecs(ITestOutputHelper helper) : base(Config, output: helper)
+        {
+            _serilogLogger = new LoggerConfiguration()
+                .WriteTo.ColoredConsole()
+                .MinimumLevel.Information()
+                .CreateLogger();
+
+            _loggingAdapter = new SerilogLoggingAdapter(Sys.Log);
+        }
+
+        [Theory]
+        [InlineData(LogLevel.DebugLevel, "test case {0}", new object[]{ 1 })]
+        [InlineData(LogLevel.DebugLevel, "test case {myNum}", new object[] { 1 })]
+        public void ShouldHandleSerilogFormats(LogLevel level, string formatStr, object[] args)
+        {
+            Sys.EventStream.Subscribe(TestActor, typeof(LogEvent));
+
+            Action logWrite = () =>
+            {
+                _loggingAdapter.Log(level, formatStr, args);
```

Comment:
So we are using a Serilog-enabled logging adapter here. It _should_ be formatting all of the strings in advance of injecting that content into the underlying `Sys.Log` logging adapter, but that doesn't appear to be the case. The standard out logger explodes behind the scenes and logs an error directly out to the console here: https://github.com/akkadotnet/akka.net/blob/3f924fda8d6ca55c5af90a710d23be8ce36d5c7a/src/core/Akka/Event/StandardOutLogger.cs#L138

I force the exception to get thrown again in the `logEvent.ToString()` call as well. It seems to me like we should be bringing the `SerilogMessageFormatter` back - that's what is supposed to be responsible for formatting these logevents safely. The `SerilogLogger` doesn't directly call the `logEvent.ToString()` message directly, if I recall correctly, so that's why it's able to handle these messages safely.

I don't think it's good design, in its current form, to have the `SerilogLoggingAdapter` produces events that can be subscribed by ANYTHING inside the an `ActorSystem` which can throw errors on a base method like `ToString`. Therefore, all of the `LogMessage` instances created by the `SerilogLoggingAdapter` should be adaptable to run in non-Serilog environments for safety purposes. I think I have a way of doing this, so I'm going to give that a try on the next commit.

Reply:
You are correct. I've been reviewing the old PR that removed the `SerilogLoggingAdapter` and I realized that I didn't took into account, that the logmessages can be consumed by anyone, not just serilog. Which is rather obvious when you point it out. So yeah, the return of the `SerilogLoggingAdapter` seems like a must.